monai/apps/generation/maisi/networks/autoencoderkl_maisi.py

```diff
@@ -0,0 +1,901 @@
+        debug: Whether to print debug information.
+    """
+
+    def __init__(
+        self,
+        num_groups: int,
+        num_channels: int,
+        eps: float = 1e-5,
+        affine: bool = True,
+        norm_float16: bool = False,
+        debug: bool = True,
+    ):
+        super().__init__(num_groups, num_channels, eps, affine)
+        self.norm_float16 = norm_float16
+        self.debug = debug
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        if self.debug:
+            print("MaisiGroupNorm3D in", input.size())
+
+        if len(input.shape) != 5:
+            raise ValueError("Expected a 5D tensor")
+
+        param_n, param_c, param_d, param_h, param_w = input.shape
+        input = input.view(param_n, self.num_groups, param_c // self.num_groups, param_d, param_h, param_w)
+
+        inputs = []
+        for i in range(input.size(1)):
+            array = input[:, i : i + 1, ...].to(dtype=torch.float32)
+            mean = array.mean([2, 3, 4, 5], keepdim=True)
+            std = array.var([2, 3, 4, 5], unbiased=False, keepdim=True).add_(self.eps).sqrt_()
+            if self.norm_float16:
+                inputs.append(array.sub_(mean).div_(std).to(dtype=torch.float16))
+            else:
+                inputs.append(array.sub_(mean).div_(std))
+
+        del input
+        _empty_cuda_cache()
+
+        input = (
+            torch.cat([inputs[k] for k in range(len(inputs))], dim=1)
+            if max(inputs[0].size()) < 500
+            else self._cat_inputs(inputs)
+        )
+
+        input = input.view(param_n, param_c, param_d, param_h, param_w)
+        if self.affine:
+            input.mul_(self.weight.view(1, param_c, 1, 1, 1)).add_(self.bias.view(1, param_c, 1, 1, 1))
+
+        if self.debug:
+            print("MaisiGroupNorm3D out", input.size())
+
+        return input
+
+    def _cat_inputs(self, inputs):
+        input_type = inputs[0].device.type
+        input = inputs[0].clone().to("cpu", non_blocking=True) if input_type == "cuda" else inputs[0].clone()
+        inputs[0] = 0
+        _empty_cuda_cache()
+
+        for k in range(len(inputs) - 1):
+            input = torch.cat((input, inputs[k + 1].cpu()), dim=1)
+            inputs[k + 1] = 0
+            _empty_cuda_cache()
+            gc.collect()
+
+            if self.debug:
+                print(f"MaisiGroupNorm3D cat: {k + 1}/{len(inputs) - 1}.")
+
+        return input.to("cuda", non_blocking=True) if input_type == "cuda" else input
+
+
+class MaisiConvolution(nn.Module):
+    """
+    Convolutional layer with optional debug output and custom splitting mechanism.
+
+    Args:
+        spatial_dims: Number of spatial dimensions (1D, 2D, 3D).
+        in_channels: Number of input channels.
+        out_channels: Number of output channels.
+        num_splits: Number of splits for the input tensor.
+        dim_split: Dimension of splitting for the input tensor.
+        debug: Whether to print debug information.
```

Comment:
Hi @dongyang0122 , thank you for the PR. 

The `debug` parameter is used in class init as an arg. Would it be better to make it kwarg with default set to `False`?

Also, can we use logger.debug to print the debugging info, as a replacement of `print`? Thanks!

Reply:
Hi @KumoLiu, I have updated the script with `logging.info`. Shall we resolve this conversation?

Reply:
Looks good to me for the change from `print` to logging.info. Thanks @dongyang0122 !